cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/BalancingAction.java

```diff
@@ -65,18 +66,46 @@ public BalancingAction(TopicPartition sourceTp,
   }
 
   /**
-   * Constructor for creating a balancing proposal with given topic partitions, source and destination broker id and logdir,
-   * and the topic partition of replica to swap with.
+   * Constructor for creating a balancing proposal with given topic partition, source and destination disk, and
+   * balancing action type.
    *
-   * @param sourceTp                Topic partition of the source replica.
-   * @param sourceBrokerId          Source broker id of the replica.
-   * @param sourceBrokerLogdir      Source broker logdir of the replica.
-   * @param destinationBrokerId     Destination broker id of the replica.
-   * @param destinationBrokerLogdir Destination broker logdir of the replica.
-   * @param actionType              Action type.
-   * @param destinationTp           Topic partition of the replica to swap with.
+   * @param tp              Topic partition of the replica.
+   * @param sourceDisk      Source disk of the replica.
+   * @param destinationDisk Destination disk of the replica.
+   * @param actionType      Action type.
+   */
+  public BalancingAction(TopicPartition tp,
```

Comment:
Maybe I am missing something -- but is this constructor used by any code? Does this allow creation of actions between disks of different brokers or just the same broker?

Reply:
It should be used in [here](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/AbstractGoal.java#L318), I will remove the `null` passed-in here.